enterprise/frontend/src/metabase-enterprise/transforms/pages/TransformListPage/TransformListPage.tsx

```diff
@@ -1,90 +1,206 @@
-    );
-  }, [transforms, debouncedSearchQuery]);
+  const isLoading = isLoadingCollections || isLoadingTransforms;
+  const error = collectionsError ?? transformsError;
 
-  const transformColumnDef = useMemo<ColumnDef<Transform, ReactNode>[]>(
+  const treeData = useMemo(
+    () => buildTreeData(collections, transforms),
+    [collections, transforms],
+  );
+
+  const defaultExpanded = useMemo(
+    () => getDefaultExpandedIds(targetCollectionId, targetCollection),
+    [targetCollectionId, targetCollection],
+  );
+
+  const columnDefs = useMemo<TreeTableColumnDef<TreeNode>[]>(
     () => [
       {
+        id: "name",
         accessorKey: "name",
-        meta: {
-          width: "auto",
-        },
         header: t`Name`,
-        cell: ({ getValue }) => (
-          <Group data-testid="transform-name" gap="sm">
-            <Icon name="transform" c="brand" />
-            {getValue()}
-          </Group>
+        minWidth: 320,
+        enableSorting: true,
+        cell: ({ row }) => (
+          <EntityNameCell
+            data-testid="tree-node-name"
+            icon={row.original.nodeType === "folder" ? "folder" : "transform"}
+            iconColor={
+              row.original.nodeType === "folder" ? "text-medium" : "brand"
+            }
+            name={row.original.name}
+          />
         ),
       },
       {
+        id: "updated_at",
         accessorKey: "updated_at",
-        cell: ({ getValue }) => {
-          const value = getValue() as string;
-          return value && <DateTime value={value} />;
-        },
-        meta: {
-          width: "auto",
-        },
         header: t`Last Modified`,
+        maxWidth: 260,
+        minWidth: "auto",
+        enableSorting: true,
+        sortingFn: "datetime",
+        sortDescFirst: true,
+        cell: ({ row }) =>
+          row.original.updated_at ? (
+            <DateTime value={row.original.updated_at} />
+          ) : null,
       },
       {
-        accessorFn: (transform: Transform) => transform.target.name,
+        id: "output_table",
+        accessorFn: (node) => node.target?.name ?? "",
         header: t`Output table`,
+        minWidth: 240,
+        enableSorting: true,
+        cell: ({ row }) =>
+          row.original.target?.name ? (
+            <Ellipsified>{row.original.target.name}</Ellipsified>
+          ) : null,
       },
     ],
     [],
   );
 
+  const navigateToTransform = useCallback(
+    (transformId: number) => {
+      dispatch(push(Urls.transform(transformId)));
+    },
+    [dispatch],
+  );
+
+  const treeTableInstance = useTreeTableInstance({
+    data: treeData,
+    columns: columnDefs,
+    getNodeId: (node) => node.id,
+    getSubRows: (node) => node.children,
+    defaultExpanded,
+    expanded: searchQuery ? true : undefined,
+    globalFilter: searchQuery,
+    onGlobalFilterChange: setSearchQuery,
+    globalFilterFn: (row, _columnId, filterValue) => {
```

Comment:
Should these functions be memoized? `useTreeTableInstance` memoizes a bunch of stuff, but because the dependency arrays reference these, it doesn't matter much